nodestream/pipeline/extractors/stores/aws/dynamodb_extractor.py

```diff
@@ -0,0 +1,199 @@
+
+def convert_number(value: str):
+    if '.' in value:
+        return float(value)
+    return int(value)
+
+DEFAULT_HANDLER = (convert_scalar, leave_untouched)
+
+def convert_value(dynamo_object: Dict[str, Any]):
+    attribute_type, value = DynamoObject(dynamo_object).items()
+    result_handler, type_handler = ATTRIBUTE_TYPE_DESCRIPTOR_HANDLERS.get(attribute_type, DEFAULT_HANDLER)
+    return result_handler(value, type_handler)
+
+
+ATTRIBUTE_TYPE_DESCRIPTOR_HANDLERS = {
+    "S": (convert_scalar, str),
+    "N": (convert_scalar, convert_number),
+    "B": (convert_scalar, bytes),
+    "NULL": (convert_scalar, None),
+    "BOOL": (convert_scalar, bool),
+    "SS": (convert_array, str),
+    "NS": (convert_array, convert_number),
+    "BS": (convert_array, bytes),
+    "M": (convert_map, convert_value),
+    "L": (convert_array, convert_value)
+}
+
+class DynamoRowConverter:
+    def __init__(self, item: Dict[str, Any]) -> None:
+        self.item = item
+
+    # { key: {"S": "string"}, secondary_key: { "M": {Stuff} }, tertiary_key: { "L": [Stuff] } }
+    def convert_row(self):
+        return {attribute_name: convert_value(attribute_value) for attribute_name, attribute_value in self.item.items()}
+
+
+def verify_list_consistency(value_list: List[Any]):
+    type_set = set([type(value) for value in value_list])
+    if len(type_set > 1):
+        return "L"
+    return REVERSE_ORDER_DESCRIPTOR_HANDLERS[type_set[0]] + "S"
+
+def boolean_to_dynamo_format_converter(value: None | bool):
+    return True if value is None else False
+
+def convert_value_to_dynamo_format(value: Any):
+    value_type, handler = REVERSE_ORDER_DESCRIPTOR_HANDLERS[type(value)]
+    if (value_type == "LIST"):
+        value_type = verify_list_consistency(value)
+    converted_value = handler(value)
+    return DynamoObject({value_type: converted_value})
+
+def recursive_dynamo_format_converter(value: Dict[str, Any] | List[Any]):
+    if isinstance(value, Dict):
+        return {key: convert_value_to_dynamo_format(item) for key, item in value.items()}
+    return [convert_value_to_dynamo_format(item) for item in value]
+    
+
+REVERSE_ORDER_DESCRIPTOR_HANDLERS = {
+    int: ("N", str),
+    float: ("N", str),
+    str: ("S", leave_untouched),
+    bytes:( "B", leave_untouched),
+    None: ("NULL", boolean_to_dynamo_format_converter),
+    bool: ("BOOL", leave_untouched),
+    dict: ("M", recursive_dynamo_format_converter),
+    list: ("LIST", recursive_dynamo_format_converter)
+}
+
+
+def convert_value_list_to_dynamo_format(value_list: List[Any]):
+    return_values = []
+    for value in value_list:
+        return_values.append(convert_value_to_dynamo_format(value))
+    return return_values
+
+
+class DynamoConditional:
+    @classmethod
+    def from_conditional_data(cls, conditional_data: Dict[str, Any]):
+        return cls(**conditional_data)
+    
+    def __init__(self, AttributeValueList: List[Any], ComparisonOperator: str):
```

Comment:
```suggestion
    def __init__(self, attribute_value_list: List[Any], comparison_operator: str):
```

I think the code needs too keep with standard python naming conventions to keep consistent with the rest of the code base.